status-payloads-spec.md

```diff
@@ -65,46 +65,100 @@ If a signature is not present but an author is provided by a layer below, the me
 
 ## Encoding
 
-The payload is encoded using [Transit format](https://github.com/cognitect/transit-format). Transit was chosen over JSON in order to reduce the bandwidth.
+The payload is encoded using [Protobuf](https://developers.google.com/protocol-buffers)
 
 ## Types of messages
 
 ### Message
 
-The type `Message` represents a text message exchanged between clients and is identified by the transit tag `c4`.
+The type `ChatMessage` represents a text message exchanged between clients.
 
 #### Payload
 
-Payload is a struct (a compound data type) with the following fields (order is important):
+The protobuf description is:
 
-<!-- TODO: Be more precise in struct description, a la RFC, e.g. TLS style https://tools.ietf.org/html/rfc8446 -->
+```protobuf
+message ChatMessage {
+  // Lamport timestamp of the chat message
+  uint64 clock = 1;
+  // Unix timestamps in milliseconds, currently not used as we use whisper as more reliable, but here
+  // so that we don't rely on it
+  uint64 timestamp = 2;
+  // Text of the message
+  string text = 3;
+  // Id of the message that we are replying to
+  string response_to = 4;
+  // Ens name of the sender
+  string ens_name = 5;
+  // Local chatID of the message
+  string chat_id = 6;
+
+  // The type of message (public/one-to-one/private-group-chat)
+  MessageType message_type = 7;
+  // The type of the content of the message
+  ContentType content_type = 8;
+
+  oneof payload {
+    StickerMessage sticker = 9;
+  }
+
+  enum MessageType {
+    UNKNOWN_MESSAGE_TYPE = 0;
+    ONE_TO_ONE = 1;
+    PUBLIC_GROUP = 2;
+    PRIVATE_GROUP = 3;
+    // Only local
+    SYSTEM_MESSAGE_PRIVATE_GROUP = 4;
+    }
+  enum ContentType {
+    UNKNOWN_CONTENT_TYPE = 0;
+    TEXT_PLAIN = 1;
+    STICKER = 2;
+    STATUS = 3;
+    EMOJI = 4;
+    TRANSACTION_COMMAND = 5;
+  }
+}
+```
+
+#### Payload
 
 | Field | Name | Type | Description |
 | ----- | ---- | ---- | ---- |
-| 1 | text | `string` | The text version of the message content |
-| 2 | content type | `enum` (more in [Content types](#content-types)) | See details |
-| 3 | message type | `enum` (more in [Message types](#message-types)) | See details |
-| 4 | clock | `int64` | See details |
-| 5 | timestamp | `int64` | See details |
-| 6 | content | `struct { chat-id string, text string, response-to string }` | The chat-id of the chat this message is destined to, the text of the content and optionally the id of the message it is responding to|
+| 1 | clock | `uint64` | The clock of the chat|
+| 2 | timestamp | `uint64` | The sender timestamp at message creation |
+| 3 | text | `string` | The content of the message |
+| 4 | response_to | `string` | The ID of the message replied to |
```

Comment:
would it make sense to have an ID type with restrictions to it? even if it is something like `stringid` and then elaborated on above

Reply:
I would probably keep it a string, because technically it is a string (so strings need to be used in protobuf), and then is validated (in the same way for other fields), I have always taken this as what is currently sent across the wire, but never really spent time questioning it.

Essentially, do we want this field to represent what is being sent in `protobuf` or we want it to diverge from that in the specs so that it's easier to interpret for someone reading it?

I don't have a strong opinion myself, either are fine

Reply:
Keeping it as a string here makes sense for accuracy

One idea would be to have a small section below on restriction/validation to explain additional requirements we have